src/main/java/at/ac/fhcampuswien/fhmdb/ui/MovieCell.java

```diff
@@ -9,35 +10,45 @@
 
 public class MovieCell extends ListCell<Movie> {
 
+    private final ClickEventHandler<Movie> clickEventHandler;
+
+    public MovieCell(ClickEventHandler<Movie> clickEventHandler) {
+        super();
+        this.clickEventHandler = clickEventHandler;
+    }
+
     @Override
     protected void updateItem(Movie movie, boolean empty) {
-        Node graphic;
-        MovieCellController controller;
-
-        try {
-            FXMLLoader loader = new FXMLLoader(MovieCell.class.getResource("movie-cell.fxml"));
-            graphic = loader.load();
-            controller = loader.getController();
-        } catch (IOException e) {
-            throw new RuntimeException(e);
-        }
-
         super.updateItem(movie, empty);
 
         if (empty || movie == null) {
             setText(null);
             setGraphic(null);
         } else {
-            controller.setTitle(movie.getTitle());
-            controller.setDescription(movie.getDescription());
-            controller.setGenre(movie.getGenres());
-            controller.setYear(movie.getReleaseYear());
-            controller.setRuntime(movie.getLengthInMinutes());
-            controller.setRating(movie.getRating());
-            controller.setWriters(movie.getWriters());
-            controller.setDirectors(movie.getDirectors());
-            controller.setMainCast(movie.getMainCast());
-            setGraphic(graphic);
+            Node graphic;
+            MovieCellController controller;
+
+            try {
+                FXMLLoader loader = new FXMLLoader(MovieCell.class.getResource("movie-cell.fxml"));
+                graphic = loader.load();
+                controller = loader.getController();
+
+                controller.getAddToWatchlistButton().setOnMouseClicked(mouseEvent -> clickEventHandler.onClick(getItem()));
+                controller.getRemoveFromWatchlistButton().setOnMouseClicked(mouseEvent -> clickEventHandler.onClick(getItem()));
+                controller.setTitle(movie.getTitle());
+                controller.setDescription(movie.getDescription());
+                controller.setGenre(movie.getGenres());
+                controller.setYear(movie.getReleaseYear());
+                controller.setRuntime(movie.getLengthInMinutes());
+                controller.setRating(movie.getRating());
+                controller.setWriters(movie.getWriters());
+                controller.setDirectors(movie.getDirectors());
+                controller.setMainCast(movie.getMainCast());
+                setGraphic(graphic);
```

Comment:
### Remove Button not shown in the Movie-Cell

Add this before setGraphic(graphic)
`if (isWatchlistCell) controller.addToWatchlist();`